### ISSUE TITLE
Issues with the new Adaptive Banner

### ISSUE THREAD

**Issue by atrope** [googleads/googleads-mobile-flutter#234]
Hey All!

i have an issue with the new adaptive banner.

We are trying to use it in our application and when we request the size inside a Future Builder it Always returns `411x62`
independent where the code is located.

Also when we try to request the ad, the banner most of the times is cut in half because of the height it was given

Here the code we are using:

```

FutureBuilder<AnchoredAdaptiveBannerAdSize?>(
      future: AdSize.getAnchoredAdaptiveBannerAdSize(
        Orientation.landscape,
        context.mediaQuerySize.width.truncate(),
      ),
      builder: (BuildContext context,
          AsyncSnapshot<AnchoredAdaptiveBannerAdSize?> snapshot) {
        if (!snapshot.hasData) {
          return Container();
        }
        final sizes = [snapshot.data ?? AdSize.largeBanner]; // Return the 
        return Container(
          constraints: BoxConstraints(
            minHeight: sizes.first.height.toDouble() + 10,
          ),
          child: _getBanner(sizes)       //here we call PublisherBannerAd and return the AdWidget
);
      },
    );
```

Should we do something else?

Thanks!

**Comment by atrope** [comment]
@bparrishMines 

Ok I Just Tried with the new Update from yesterday(and the example) and now it is still cutting but not always..

Here you can see an ad that is cut:

![screenshot-1621355170328](https://user-images.githubusercontent.com/2945478/118688678-a55dad00-b7dc-11eb-9e2a-7ba5b09d6153.jpg)


Also I saw that you are not supporting PublisherBannerAd i try to cast ad to PublisherBannerAd and it gives me the error:

`type 'PublisherBannerAd' is not a subtype of type 'BannerAd?' in type cast`

Is there any reason not to support it?

Thanks

**Comment by bparrishMines** [comment]
@atrope I'm assuming you're referring to the Ad with the shoe that is cut off. Are you passing the full width of the device when retrieving the AdSize. According to [Adaptive Banner Implementation Notes](https://developers.google.com/admob/android/banner/adaptive#implementation_notes), they work best when 

> The adaptive banner sizes are designed to work best when using the full available width. In most cases, this will be the full width of the screen of the device in use. Be sure to take into account applicable display cutouts.

Also, when are you casting to `PublisherBannerAd`. In the `AdListener` callback or after instantiating a `BannerAd`?

**Comment by atrope** [comment]
Hey @bparrishMines  yes that's the on, sorry for not clearly pointing that out.

yes i am passing the full width as your example shows.

I Am also following your example and casting it after it has loaded and getting the width/height after setting the banner.

**Comment by bparrishMines** [comment]
Could you verify that `snapshot.data` is not retuning `null`? The `Ad` looks as if it wants to show a large banner, but the height is set to the `AdaptiveBannerAdSize` height. But, I'm not sure how this would happen.

Also, what `Ad` is being created in `_getBanner`. I don't think I would be able to debug this without having a small sample of what is being done in that method.

**Comment by atrope** [comment]
hey @bparrishMines  i have changed the code to reflect 100% what you've done in your example.

I will try to produce a sample and put it here.

**Comment by atrope** [comment]
Hello @bparrishMines With 0.13 the Casting problem was solved 🥳 

As for the cut banner, I talked with my google rep and she directed me to this article:
https://support.google.com/admanager/answer/9464128

I had to create a new Ad Unit and LI on Ad Manger for this Adaptive banner to work.
Maybe we should have this banner documented when referring to Adaptive Banners?

Also i was confusing Adaptive Anchored Banners with Adaptive Inline Banners that would also provide InLine adaptive banners (I Guess those will be coming soon :))
I Guess This issue can be closed :)

**Comment by atrope** [comment]
Hello @bparrishMines With 0.13 the Casting problem was solved 🥳 

As for the cut banner, I talked with my google rep and she directed me to this article:
https://support.google.com/admanager/answer/9464128

I had to create a new Ad Unit and LI on Ad Manger for this Adaptive banner to work.
Maybe we should have this banner documented when referring to Adaptive Banners?

Also i was confusing Adaptive Anchored Banners with Adaptive Inline Banners that would also provide InLine adaptive banners (I Guess those will be coming soon :))

I Only realized that there is one problem.

Is it possible to return the Loaded Banner size? 
I Know we can get the request size via the "banner.sizes" but i was looking for the actual rendered banner size.
